test/jdk/java/awt/dnd/DragToAnotherScreenTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2005, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Label;
+import java.awt.Toolkit;
+import java.awt.Window;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.awt.dnd.DnDConstants;
+import java.awt.dnd.DragGestureListener;
+import java.awt.dnd.DragSource;
+import java.awt.dnd.DropTarget;
+import java.awt.dnd.DropTargetAdapter;
+import java.awt.dnd.DropTargetDropEvent;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 6179157
+ * @summary Tests dnd to another screen
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual DragToAnotherScreenTest
+ */
+
+public class DragToAnotherScreenTest {
+    private static Label label0;
+    private static Label label1;
+    private static final int HGAP = 20;
+
+    private static final String INSTRUCTIONS = """
+                The following test is applicable for Single as well
+                as Multi-monitor screens.
+
+                If on multi-monitor screens then please position
+                the drag and drop windows on different screens.
+
+                If you can not move the mouse from the frame "Drag Source"
+                to the frame "Drop Target" press PASS.
+
+                Otherwise drag the label "Drag me" and
+                drop it on the label "Drop on me".
+
+                If you can not drag to the second label (for example
+                if you can not drag across screens) press FAIL.
+
+                If the second label changes its text to drag me
+                after the drop and you DO NOT see any error messages
+                in the log area press PASS else FAIL.
```

Comment:
I think we can make it semi-automatic then. If the test determines an error condition, the test could fail automatically.

However, to give a better experience for the tester, the test could display a message before calling `forceFail`.

Another way could be disabling the `Pass` button, yet it's not possible at the moment.

Reply:
@aivanov-jdk I have added PFJ.forcePass() and forceFail() , added JOptionPane msg so that user has time to read the success/failure msg. 

Just a thought: May be it it is good to add a delay in PFJ's forcePass()  and forceFail() which gives time for the user to see any messages or logs in the logArea.

Reply:
> Just a thought: May be it it is good to add a delay in PFJ's forcePass() and forceFail() which gives time for the user to see any messages or logs in the logArea.

I don't think it's good. What if there's no log area?

Adding a helper method to display message sounds reasonable though. The message could time out.

> Another way could be disabling the *Pass* button, yet it's not possible at the moment.

There's a bug for this feature [JDK-8317111](https://bugs.openjdk.org/browse/JDK-8317111): *Provide access to Pass/Fail buttons of PassFailJFrame*.

Reply:
> Adding a helper method to display message sounds reasonable though. The message could time out.
Yes, something like a JOptionPane which displays a msg and auto-closes after a 5 sec count down.